modules/clickhouse/clickhouse_test.go

```diff
@@ -0,0 +1,274 @@
+package clickhouse
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	ch "github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+const dbname = "testdb"
+const user = "clickhouse"
+const password = "password"
+
+const createTableQuery = "create table if not exists test_table (id UInt64) engine = MergeTree PRIMARY KEY (id) ORDER BY (id) SETTINGS index_granularity = 8192;"
+const insertQuery = "INSERT INTO test_table (id) VALUES (1);"
+const selectQuery = "SELECT * FROM test_table;"
+
+type Test struct {
+	Id uint64
+}
+
+func TestClickHouseDefaultConfig(t *testing.T) {
+	ctx := context.Background()
+
+	container, err := RunContainer(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Clean up the container after the test is complete
+	t.Cleanup(func() {
+		assert.NoError(t, container.Terminate(ctx))
+	})
+
+	connectionHost, err := container.ConnectionHost(ctx)
+	assert.NoError(t, err)
+
+	conn, err := ch.Open(&ch.Options{
+		Addr: []string{connectionHost},
+		Auth: ch.Auth{
+			Database: container.dbName,
+			Username: container.user,
+			Password: container.password,
+		},
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, conn)
+	defer conn.Close()
+
+	err = conn.Ping(context.Background())
+	assert.NoError(t, err)
+}
+
+func TestClickHouseIpPort(t *testing.T) {
```

Comment:
First of all, thank you for your comments 🙏

You are right, `TestClickHouseConnectionHost` would be better function name. I just wanted to be clear about `host` means `ip:port` but not needed I guess.

I left duplicate crud operations on purpose. I just want to show that all methods works in same way with same operations. But I am not sure if it is necessary actually. Maybe it would better to create another test for CRUD operations just like you mentioned next comment. I am a little confused there 😅